src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AddColumns.cs

```diff
@@ -86,33 +89,43 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
                 // type analysis, so they need to be known upfront (before AddColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String ||
-                    (nameNode = nameArg.AsStrLit()) == null)
+                StrLitNode strLitNode = null;
+                IIdentifierNode identifierNode = null;
+
+                if ((!supportIndentifiers && (nameArgType.Kind != DKind.String || (strLitNode = nameArg.AsStrLit()) == null)) ||
+                     (supportIndentifiers && (nameArgType.Kind != DKind.Identifier || (identifierNode = nameArg.AsIdentifierNode()) == null)))
                 {
                     fArgsValid = false;
+
+                    // Argument '{0}' is invalid, expected a text literal.
                     errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
                     continue;
                 }
```

Comment:
I'd leave this intact and just say:

string value;
if (!supportIndentifiers) { 
   // existing code, unchanged if
  value = nameNode.Value;
} else {
  // new code, 
  // verify it's a FirstNameNode (See SetFunction for example)
  value = nameNode.AsFirstNameNode().Ident.Name;
}

Reply:
Not sure why